### PR TITLE
Add AMD GPU extended resource test

### PR DESCRIPTION
**Pull request by engedaam**
Fixes # 

**Description**
Added new integration test for AMD GPU Extended Resource 

**How was this change tested?**

*

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/suites/integration/extended_resources_test.go [hunk 1]
```diff
@@ -158,6 +163,54 @@ var _ = Describe("Extended Resources", func() {
...
+					},
+					Limits: v1.ResourceList{
+						"amd.com/gpu": resource.MustParse("1"),
+					},
+				},
+			},
+		})
+		selector := labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+		env.ExpectCreated(provisioner, provider, dep)
+		EventuallyWithOffset(1, func(g Gomega) {
```
Comment (jonathan-innis) [inline test/suites/integration/extended_resources_test.go:208]: Why not use `ExpectPodsHealthy`?

**Review by jonathan-innis — COMMENTED** [review]
Great work! This was no small task!

File: website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml [hunk 2]
```diff
@@ -73,6 +73,13 @@ Resources:
             Action:
               - iam:PassRole
             Resource: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/KarpenterNodeRole-${ClusterName}"
+          -  Effect: Allow
+             Action:
+              - s3:s3:Get*
+              - s3:List*
+              - s3-object-lambda:Get*
+              - s3-object-lambda:List*
+             Resource: "*"
```
Comment (ellistarn) [inline website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml:82]: Why do we need this?

Comment (engedaam) [reply website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml:82]: To be able to download the AMD Driver the node needs to be able to have [AmazonS3ReadOnlyAccess](https://us-east-1.console.aws.amazon.com/iam/home#/policies/arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess)

File: test/suites/integration/extended_resources_test.go (hunk 1 above)
Comment (engedaam) [reply test/suites/integration/extended_resources_test.go:208]: The driver installation requires the Timeout value to be extended. The `ExpectPodsHealthy` does not allow for that change.

File: website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml (hunk 2 above)
Comment (ellistarn) [reply website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml:82]: We definitely don't want to give these permissions to users going through this getting started guide.

Comment (FernandoMiguel) [reply website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml:82]: I would very much be upset if all our nodes had full read access to all S3 buckets in their AWS Account.

Comment (engedaam) [reply website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml]: Removed the read access.

File: test/suites/integration/extended_resources_test.go [hunk 3]
```diff
@@ -158,6 +163,54 @@ var _ = Describe("Extended Resources", func() {
 		env.ExpectCreatedNodeCount("==", 1)
 		env.EventuallyExpectCreatedNodesInitialized()
 	})
+	It("should provision nodes for a deployment that requests amd.com/gpu", func() {
+		ExpectAMDDevicePluginCreated()
+		DeferCleanup(func() {
+			ExpectAMDDevicePluginDeleted()
+		})
+
+		content, err := os.ReadFile("testdata/amd_driver_input.golden")
```
Comment (ellistarn) [inline test/suites/integration/extended_resources_test.go:172]: I'm not sure I understand golden in this context. Typically .golden is referred to as something that's checked, not something that's used.

File: test/suites/integration/testdata/amd_driver_input.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+MIME-Version: 1.0
```
Comment (ellistarn) [inline test/suites/integration/testdata/amd_driver_input.txt:1]: Should this be a .sh? Does it make sense to align the other files in this directory too, while we're in here?

File: test/suites/integration/extended_resources_test.go (hunk 3 above)
Comment (engedaam) [reply test/suites/integration/extended_resources_test.go]: Adjusted the file name

File: test/suites/integration/testdata/amd_driver_input.txt (hunk 4 above)
Comment (engedaam) [reply test/suites/integration/testdata/amd_driver_input.txt:1]: I can do that

File: test/suites/integration/extended_resources_test.go [hunk 5]
```diff
@@ -239,6 +292,88 @@ func ExpectNvidiaDevicePluginDeleted() {
 	})
 }
 
+func ExpectAMDDevicePluginCreated() {
+	env.ExpectCreatedWithOffset(1, &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "amdgpu-device-plugin-daemonset",
+			Namespace: "kube-system",
```
[the remark this answers is not in the document]
Comment (engedaam) [reply test/suites/integration/extended_resources_test.go:284]: Updated!

**Review by ellistarn — APPROVED** [review]

**Pull request merged**